insights/parsers/gnocchi.py

```diff
@@ -0,0 +1,109 @@
+"""
+OpenStack gnocchi related files or commands
+===========================================
+
+Module for the processing of gnocchi related files and commands.
+This module provides parsers:
+
+GnocchiConf - file ``/etc/gnocchi.gnocchi.conf``
+------------------------------------------------
+
+GnocchiMetricdLog - file ``gnocchi-metricd.log`` or ``metricd.log``
+-------------------------------------------------------------------
+"""
+
+from .. import parser, IniConfigFile, LogFileOutput
+from insights.specs import Specs
+
+
+@parser(Specs.gnocchi_conf)
+class GnocchiConf(IniConfigFile):
+    """
+    Gnocchi configuration parser class, based on the ``IniConfigFile`` class.
+
+    The gnocchi configuration file is a standard '.ini' file and this parser uses
+    the ``IniConfigFile`` class to read it.
+
+    Sample configuration::
+
+        [DEFAULT]
+        log_dir = /var/log/gnocchi
+        [api]
+        auth_mode = keystone
+        max_limit = 1000
+        [archive_policy]
+        [indexer]
+        url = mysql+pymysql://gnocchi:exampleabckeystring@192.168.0.1/gnocchi?charset=utf8
+        [metricd]
+        workers = 2
+        [oslo_middleware]
+        [oslo_policy]
+        policy_file = /etc/gnocchi/policy.json
+        [statsd]
+        resource_id = 5e3fcbe2-7aab-475d-b42c-abcdefgh
+        user_id = e0ca4711-1128-422c-abd6-abcdefgh
+        project_id = af0c88e8-90d8-4795-9efe-abcdefgh
+        archive_policy_name = high
+        flush_delay = 10
+        [storage]
+        driver = file
+        file_basepath = /var/lib/gnocchi
+        [keystone_authtoken]
+        auth_uri=http://192.168.0.1:5000/v3
+        auth_type=password
+        auth_version=v3
+        auth_url=http://192.168.0.1:35357
+        username=gnocchi
+        password=yourpassword23432
+        user_domain_name=Default
+        project_name=services
+        project_domain_name=Default
+
+
+    Examples:
+
+        >>> conf = shared[GnocchiConf]
```

Comment:
`shared` is no longer used in 3.x. please modify the examples as per 3.0

Reply:
Hi @chenlizhong Sorry for jumping in but I see in some of parsers we have started including doctest. Please refer https://github.com/RedHatInsights/insights-core/pull/1092/files#diff-340ae940da9a54f85293da193efe7a17R54. This is a good time to include those in this parser as well.